### PR TITLE
Feat/hp/room2

### PR DESCRIPTION
**Pull request by hrishikeshpaul**
Try it out!

### CODE REVIEW

File: vb-be/src/modules/player/player.controller.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  Query,
+  Headers,
+  Get,
+  Post,
+} from '@nestjs/common';
+import { Response as Res } from 'express';
+
+import { HttpStatus } from 'src/util/http';
+
+import { firstValueFrom } from 'rxjs';
```
Comment (s-munro) [inline vb-be/src/modules/player/player.controller.ts:11]: fix imports im gonna kill u lol

Comment (hrishikeshpaul) [reply vb-be/src/modules/player/player.controller.ts:11]: fooooook

File: vb-be/src/modules/room/room.controller.ts [hunk 2]
```diff
@@ -1,19 +1,29 @@
...
 
 import { HttpStatus } from 'src/util/http';
 import { ErrorHandler, ErrorText } from 'src/util/error';
 
 import { RoomService } from '@modules/room/room.service';
 import { ICreateRoom } from '@modules/room/room.constants';
 import { TagService } from '@modules/tag/tag.service';
-import { Types } from 'mongoose';
+import { SpotifyService } from '@modules/spotify/spotify.service';
+import { firstValueFrom } from 'rxjs';
```
Comment (s-munro) [inline vb-be/src/modules/room/room.controller.ts]: meep meep

File: vb-be/src/modules/room/room.controller.ts [hunk 3]
```diff
@@ -43,4 +53,22 @@ export class RoomController {
...
+  ) {
+    try {
+      const spotifyAT = req.headers['v-s-at'] as string;
+      const response = await firstValueFrom(
+        this.spotify.getUserPlaylists(spotifyAT, offset),
+      );
+      return res.status(HttpStatus.OK).json(response.data);
+    } catch (err) {
+      console.log(err);
+      // return res.status(err.statusCode).send(err);
```
Comment (s-munro) [inline vb-be/src/modules/room/room.controller.ts]: nit: looks like you may have forgotten to uncomment this

File: vb-be/src/modules/socket/socket.gateway.ts [hunk 4]
```diff
@@ -12,36 +12,50 @@ import { HttpStatus } from 'src/util/http';
 
 import { RoomService } from '@modules/room/room.service';
 import { ISocketCreateRoomData } from '@modules/socket/socket.constants';
+import { WsGuard } from './socket.middleware';
```
Comment (s-munro) [inline vb-be/src/modules/socket/socket.gateway.ts]: nit relative

Comment (hrishikeshpaul) [reply vb-be/src/modules/socket/socket.gateway.ts]: this file is a big WIP lol

File: vb-be/src/modules/spotify/spotify.constants.ts [hunk 5]
```diff
@@ -43,3 +43,120 @@ export interface SpotifyPublicUser {
   uri: string;
   email: string;
 }
+
+export type Restrictions = {
```
Comment (s-munro) [inline vb-be/src/modules/spotify/spotify.constants.ts:47]: she a thicc types file

Comment (hrishikeshpaul) [reply vb-be/src/modules/spotify/spotify.constants.ts:47]: i got this from somewhere

File: vb-be/src/modules/user/user.service.ts [hunk 6]
```diff
@@ -1,6 +1,7 @@
 import { Injectable } from '@nestjs/common';
 
 import { UserModel, UserType, IUser } from '@modules/user/user.schema';
+import { Types } from 'mongoose';
```
Comment (s-munro) [inline vb-be/src/modules/user/user.service.ts]: anotha one

File: vb-be/src/modules/spotify/spotify.service.ts [hunk 7]
```diff
@@ -99,4 +100,80 @@ export class SpotifyService {
       },
     });
   }
+
```
Comment (s-munro) [inline vb-be/src/modules/spotify/spotify.service.ts:103]: services looking super solid

File: vb-fe/.eslintrc.json [hunk 8]
```diff
@@ -56,7 +56,7 @@
         }
       ],
       "import/no-unresolved": "error",
-      "indent": [2, "tab"],
+      "indent": [4, "tab"],
```
Comment (s-munro) [inline vb-fe/.eslintrc.json]: no way jose we have beef here

File: vb-fe/src/_store/player/PlayerActions.ts [hunk 9]
```diff
@@ -0,0 +1,60 @@
+import { Dispatch } from "redux";
```
Comment (s-munro) [inline vb-fe/src/_store/player/PlayerActions.ts:1]: again young blood

File: vb-fe/.eslintrc.json (hunk 8 above)
Comment (hrishikeshpaul) [reply vb-fe/.eslintrc.json]: my shit wasn't compiling!!!! that was the only fix lol!!!!!

File: vb-fe/src/components/player/PlayerVolume.tsx [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+export const PlayerVolume: FunctionComponent = (): JSX.Element => {
+  const [debounce] = useDebounce();
+  const [volume, setVolume] = useState<number | null>(null);
+
+  useEffect(() => {
+    WebPlayer.getPlayer()
+      .getVolume()
+      .then((vol: number) => {
+        setVolume(vol * 100);
+        console.log(vol, volume);
```
Comment (s-munro) [inline vb-fe/src/components/player/PlayerVolume.tsx]: nit: console.log in FE

**Review by s-munro — CHANGES_REQUESTED** [review]
Sheeesh that was a big one.  Looks great to me!  I didn't see any logic that looked suspicious or anything - just be sure to fix all of the imports/comments/console.logs/unused imports and then everything should be good to merge.  I pinged a couple of imports, but there are more throughout, so just give a look once-through and then it should be good 👍

File: vb-fe/.eslintrc.json (hunk 8 above)
Comment (s-munro) [reply vb-fe/.eslintrc.json]: No way lmao, how weird.  All good with me, as long as the BE is on 2

**Comment by hrishikeshpaul** [comment]
Cooool! the latest commit fixes all of these

**Comment by s-munro** [comment]
Screen Shot 2021-08-12 at 7 20 45 PM

I got this error while trying to spin things up and open a room.  I faced this the other day too in certain conditions, any familiarity with it?

It also seems like visiting the /room/:id route while unauthenticated doesn't push you to the `/login`.  I see that we have it listed under `AuthenticatedApp` though, so I'm looking into if we're able to implement this.

**Comment by hrishikeshpaul** [comment]
Hmmm idk why there was that error. But the unauthenticated happens cause the sockets don't have the refresh mechanism. The push to /login is for axios. It's a little annoying whole developing yeah. I was wondering if we could do the sockets refresh thing asap

**Comment by hrishikeshpaul** [comment]
@s-munro is this good to merge?

**Comment by s-munro** [comment]
> @s-munro is this good to merge?

Reviewing again now

**Comment by hrishikeshpaul** [comment]
Just pushed a few more changes!

**Comment by s-munro** [comment]
Alright lets pull the trigger on this 👍

**Comment by hrishikeshpaul** [comment]
SHEESH

**Pull request merged**